runtime/src/accounts.rs

```diff
@@ -1011,21 +1007,13 @@ pub fn prepare_if_nonce_account(
     tx_result: &Result<()>,
     maybe_nonce_rollback: Option<(&Pubkey, &AccountSharedData, Option<&AccountSharedData>)>,
     last_blockhash_with_fee_calculator: &(Hash, FeeCalculator),
-    fix_recent_blockhashes_sysvar_delay: bool,
 ) -> bool {
     if let Some((nonce_key, nonce_acc, _maybe_fee_account)) = maybe_nonce_rollback {
         if account_pubkey == nonce_key {
-            let overwrite = if tx_result.is_err() {
+            if tx_result.is_err() {
                 // Nonce TX failed with an InstructionError. Roll back
                 // its account state
                 *account = nonce_acc.clone();
-                true
-            } else {
```

Comment:
There's a change in behavior here.  Pre-code overwrites the account state regardless of TX success, post-code only on error.  This _should_ be fine, as the runtime values are the same as what the system program updated the state with in the success case.  I'm not sure we want to make that optimization(?) in the same change set?

Reply:
Scratch that. Can't read